cli/base-command.ts

```diff
@@ -253,95 +253,41 @@ export class UniswapSORCLI extends Command {
           new EIP1559GasPriceProvider(provider)
         ),
         gasModelFactory: new HeuristicGasModelFactory(),
-        tokenProvider: tokenProvider,
+        tokenProvider: this.tokenProvider,
       });
-    }
-
-    let swapRoutes: SwapRoute<any> | null;
-    if (exactIn) {
-      const amountIn = parseAmount(amountStr, tokenIn);
-      //const amountIn = CurrencyAmount.fromRawAmount(tokenIn, amountStr);
-      swapRoutes = await router.routeExactIn(
-        tokenIn,
-        tokenOut,
-        amountIn,
-        {
-          deadline: 100,
-          recipient,
-          slippageTolerance: new Percent(5, 10_000),
-        },
-        {
-          topN,
-          topNTokenInOut,
-          topNSecondHop,
-          topNWithEachBaseToken,
-          topNWithBaseToken,
-          topNWithBaseTokenInSet,
-          maxSwapsPerPath,
-          minSplits,
-          maxSplits,
-          distributionPercent,
-        }
-      );
-    } else {
-      const amountOut = parseAmount(amountStr, tokenOut);
-      swapRoutes = await router.routeExactOut(
-        tokenIn,
-        tokenOut,
-        amountOut,
-        {
-          deadline: 100,
-          recipient,
-          slippageTolerance: new Percent(5, 10_000),
-        },
-        {
-          topN,
-          topNTokenInOut,
-          topNSecondHop,
-          topNWithEachBaseToken,
-          topNWithBaseToken,
-          topNWithBaseTokenInSet,
-          maxSwapsPerPath,
-          minSplits,
-          maxSplits,
-          distributionPercent,
-        }
-      );
-    }
 
-    if (!swapRoutes) {
-      log.error(
-        `Could not find route. ${
-          debug ? '' : 'Run in debug mode for more info'
-        }.`
-      );
-      return;
-    }
+      this.id == 'quote-to-ratio'
+        ? (this._swapToRatioRouter = router)
+        : (this._router = router);
+      }
```

Comment:
Nit: It would be fine to just initialize both, not really any overhead or perf hit